pkg/kgo/client.go

```diff
@@ -86,7 +86,12 @@ func stddial(ctx context.Context, addr string) (net.Conn, error) {
 }
 
 // NewClient returns a new Kafka client with the given options or an error if
-// the options are invalid.
+// the options are invalid. The connection is initiated lazily. Upon the first
```

Comment:
Maybe `The connection is initiated lazily. Upon the first... if needed).` => `Connections to brokers are lazily created only when requests are written to them.`?